### ISSUE TITLE
specify that Backdrop will not work with airplane mode

### ISSUE THREAD

**Issue by pdewouters** [humanmade/Backdrop#9]
Took me a while to realize that Backdrop won't trigger tasks when the "Airplane Mode" plugin is enabled. Probably due to the call to `wp_remote_post`?
Would be good to make a note of it.

**Comment by rmccue** [comment]
Seems like Airplane Mode should allow loopback connections; likely breaking built-in cron otherwise as well.

cc @norcross, any insights?

**Comment by norcross** [comment]
there's a filter to allow specific ones to go through. it's there since most calls go to a website which cannot be reached (due to, ya know, airplane mode)

**Comment by rmccue** [comment]
Right, but by default, I'd think it should allow HTTP requests to the site's domain; WP's built-in cron stuff uses the same thing, as does @markjaquith's TLC Transients. I dunno, am I crazy? :)

**Comment by norcross** [comment]
I guess it never came up before? the cron stuff is iffy because some are a pass through to external sites (if I remember correctly, could be totally wrong)

how would you implement in Airplane?

**Comment by rmccue** [comment]
> how would you implement in Airplane?

The same logic that's used with the `WP_HTTP_BLOCK_EXTERNAL` constant is a good start (although using the constant directly is probably a non-starter, per norcross/airplane-mode#27; I'm surprised core doesn't use a filter there).

Specifically:

```
$check = parse_url($uri);
if ( ! $check )
    return true;

$home = parse_url( get_option('siteurl') );

// Don't block requests back to ourselves by default.
if ( 'localhost' == $check['host'] || ( isset( $home['host'] ) && $home['host'] == $check['host'] ) ) {
    /**
     * Filter whether to block local requests through the proxy.
     *
     * @since 2.8.0
     *
     * @param bool $block Whether to block local requests through proxy.
     *                    Default false.
     */
    return apply_filters( 'block_local_requests', false );
}
```

**Comment by norcross** [comment]
lemme play with it and see what works.
